sponge/src/main/net/buycraft/plugin/sponge/BuycraftPlugin.java

```diff
@@ -0,0 +1,276 @@
+    @Getter
+    private final BuycraftConfiguration configuration = new BuycraftConfiguration();
+    @Getter
+    private ServerInformation serverInformation;
+    @Getter
+    private KeenClient keenClient;
+    @Getter
+    private OkHttpClient httpClient;
+    @Getter
+    private IBuycraftPlatform platform;
+    @Getter
+    private CommandExecutor commandExecutor;
+    @Inject
+    @Getter
+    private Logger logger;
+    @Inject
+    @Getter
+    @ConfigDir(sharedRoot = false)
+    private Path workFolder;
+    @Getter
+    private boolean running = true;
+    @Getter
+    private ListingUpdateTask listingUpdateTask;
+
+    @Listener
+    public void onGamePreInitialization(GamePreInitializationEvent event) {
+        platform = new SpongeBuycraftPlatform(this);
+
+        try {
+            Files.createDirectories(workFolder);
+
+            Path configPath = workFolder.resolve("config.properties");
+            if (!configPath.toFile().exists()) {
+                configuration.fillDefaults();
+                configuration.save(configPath);
+            } else {
+                configuration.load(workFolder.resolve("config.properties"));
+                configuration.fillDefaults();
+            }
+        } catch (IOException e) {
+            getLogger().error("Unable to load configuration!", e);
+            running = false;
+            return;
+        }
+
+        httpClient = new OkHttpClient.Builder()
+                .connectTimeout(500, TimeUnit.MILLISECONDS)
+                .writeTimeout(1, TimeUnit.SECONDS)
+                .readTimeout(3, TimeUnit.SECONDS)
+                .build();
+
+        String serverKey = configuration.getServerKey();
+        if (serverKey == null || serverKey.equals("INVALID")) {
+            getLogger().info("Looks like this is a fresh setup. Get started by using 'buycraft secret <key>' in the console.");
+        } else {
+            getLogger().info("Validating your server key...");
+            ApiClient client = new ProductionApiClient(configuration.getServerKey(), httpClient);
+            try {
+                updateInformation(client);
+            } catch (IOException | ApiException e) {
+                getLogger().warn(String.format("We can't check if your server can connect to Buycraft: %s", e.getMessage()));
+            }
+            apiClient = client;
+        }
+
+        placeholderManager.addPlaceholder(new NamePlaceholder());
+        placeholderManager.addPlaceholder(new UuidPlaceholder());
+
+        platform.executeAsyncLater(
+                duePlayerFetcher = new DuePlayerFetcher(platform),
+                20,
+                TimeUnit.MILLISECONDS
+        );
+
+        commandExecutor = new QueuedCommandExecutor(platform);
+
+        Sponge.getScheduler().createTaskBuilder()
+                .execute((Runnable) commandExecutor)
+                .delay(1, TimeUnit.MILLISECONDS)
+                .interval(1, TimeUnit.MILLISECONDS)
```

Comment:
Ticks not supported!

![image](https://cloud.githubusercontent.com/assets/3318873/13483735/d45e6338-e0ff-11e5-9da4-f9bfd8a52e82.png)